src/widget/systemtrayicon.cpp

```diff
@@ -95,28 +95,12 @@ SystemTrayIcon::SystemTrayIcon()
     }
     #endif
     #ifdef ENABLE_SYSTRAY_STATUSNOTIFIER_BACKEND
-    else if (desktop == "kde"
-            && getenv("KDE_SESSION_VERSION") == QString("5"))
+    else if (desktop == "kde" && getenv("KDE_SESSION_VERSION") == QString("5"))
     {
-        qDebug() << "Using Status Notifier backend";
-        backendType = SystrayBackendType::StatusNotifier;
-        gtk_init(nullptr, nullptr);
-        snMenu = gtk_menu_new();
-        void (*callbackFreeImage)(guchar*, gpointer) =
-                [](guchar*, gpointer image)
-        {
-            delete reinterpret_cast<QImage*>(image);
-        };
-        QImage* image = new QImage(":/img/icon.png");
-        if (image->format() != QImage::Format_RGBA8888_Premultiplied)
-        *image = image->convertToFormat(QImage::Format_RGBA8888_Premultiplied);
-        GdkPixbuf* pixbuf = gdk_pixbuf_new_from_data(image->bits(), GDK_COLORSPACE_RGB, image->hasAlphaChannel(),
-                                 8, image->width(), image->height(),
-                                 image->bytesPerLine(), callbackFreeImage, image);
-
-        statusNotifier = status_notifier_new_from_pixbuf("qtox",
-                            STATUS_NOTIFIER_CATEGORY_APPLICATION_STATUS, pixbuf);
-        status_notifier_register(statusNotifier);
+        qDebug() << "Using the Qt backend";
+        qtIcon = new QSystemTrayIcon;
+        backendType = SystrayBackendType::Qt;
+        connect(qtIcon, &QSystemTrayIcon::activated, this, &SystemTrayIcon::activated);
```

Comment:
That seems wrong, you're using the Qt backend inside an #ifdef ENABLE_SYSTRAY_STATUSNOTIFIER_BACKEND

The ifdef should just be removed if you don't actually use status notifier here.